lanes/mev/check_tx.go

```diff
@@ -162,17 +162,28 @@ func (handler *CheckTxHandler) CheckTx() CheckTx {
 			handler.baseApp.Logger().Info(
 				"invalid bid tx",
 				"err", err,
+				"tx", req.Tx,
+				"removing tx from mempool", true,
 			)
 
+			// attempt to remove the transaction from the mempool (if it exists)
+			handler.mevLane.Remove(tx)
```

Comment:
I would if/else on the error type here. If the error isn't a `This doesn't exist` we should log the error.